src/transformers/sinon.test.ts

```diff
@@ -293,6 +293,79 @@ describe('spies and stubs', () => {
 `
     )
   })
+
+  it('handles on*Call', () => {
+    expectTransformation(
+      `
+      import sinon from 'sinon-sandbox'
+
+      stub.onFirstCall().returns(5)
+      stub.onSecondCall().returns(6, 7)
+
+      stub.onFirstCall().returnsArg(0)
+      stub.onSecondCall().returnsArg(1)
+
+      // invalid onCall() params
+      stub.onCall().returns('invalid')
+`,
+      `
+      stub.mockImplementation(() => {
+            if (stub.mock.calls.length === 0)
+                  return 5;
+      })
+      stub.mockImplementation(() => {
+            if (stub.mock.calls.length === 1)
+                  return 6;
+      })
+
+      stub.mockImplementation((...args) => {
+            if (stub.mock.calls.length === 0)
+                  return args[0];
+      })
+      stub.mockImplementation((...args) => {
+            if (stub.mock.calls.length === 1)
+                  return args[1];
+      })
+
+      // invalid onCall() params
+      stub.onCall().mockReturnValue('invalid')
+`
+    )
+  })
+
+  it('handles on*Call (parser: ts)', () => {
+    expectTransformation(
+      `
+      import sinon from 'sinon-sandbox'
+
+      stub.onThirdCall().returns([8, 9, 10])
+      stub.onCall(3).returns(biscuits)
+
+      stub.onThirdCall().returnsArg(2)
+      stub.onCall(3).returnsArg(biscuits)
+`,
+      `
+      stub.mockImplementation(() => {
+            if (stub.mock.calls.length === 2)
+                  return [8, 9, 10];
```

Comment:
Personally I find one-line statements less readable and error prone.

It would be slightly nicer with an arrow function here.

Reply:
hey @skovhus, I'm not sure I understand what you're asking for here.

do you mean braces around the `if`?

```ts
      stub.mockImplementation(() => {
            if (stub.mock.calls.length === 2) {
                  return [8, 9, 10];
            }
```

you said "arrow function" so maybe you mean this?

```ts
      stub.mockImplementation(() => stub.mock.calls.length === 2 && [8, 9, 10]);
```

Reply:
because the second route (`stub.mockImplementation(() => stub.mock.calls.length === 2 && [8, 9, 10]);`) would sometimes produce false for a return value (which differs from sinon)

```js
const sinon = require('sinon');

const s = sinon.stub();

s.onFirstCall().returns(5);

console.log(s());  // 5
console.log(s());  // undefined
```

 i went with braces around the single-line if statements.  hopefully that's what you meant.

Reply:
Yes I would prefer us to either use block statements or arrow functions instead of the one-line statements. Thanks for fixing this.